website/pages/docs/k8s/connect/ingress-gateways.mdx

```diff
@@ -5,4 +5,174 @@ sidebar_title: Ingress Gateways
+Values in this section override the values from the defaults stanza for the given ingress gateway with one exception.
+The annotations from the defaults stanza will be appended to any user-defined annotations defined in the gateways stanza.
+Please refer to the ingress gateway configuration [documentation](https://www.consul.io/docs/k8s/helm#v-ingressgateways-defaults) for a detailed explanation of each option.
+
+-> *Note*: Make sure any ports that will be used as listeners in the ingress gateway's Consul config entry are included in the `ports` object for each gateway.
+
+## Deploying the helm chart
+
+Ensure you have the latest consul-helm chart by running the following command:
+
+```shell-session
+ $ helm repo add hashicorp https://helm.releases.hashicorp.com
+```
+
+Assuming you saved your configuration to a file called config.yaml in the local directory, you can apply your configuration with the following command:
+
+```shell-session
+$ helm install -f config.yaml <consul-release> hashicorp/consul --wait
+```
+
+## Registering a service with Consul
+
+Now that Kubernetes has been updated, you must add the corresponding configuration to Consul. This requires you to use the consul cli.
+Registering a service with Consul is a multi-step process that consists of:
+
+* Accessing the Consul agent
+* Submitting an Ingress Gateway configuration entry to Consul
+
+### Accessing the Consul agent
+
+You can access the Consul server directly from your host via `kubectl port-forward`, this is helpful for interacting with the consul-ui locally as well as validating connectivity of the application.
+
+```shell-session
+$ kubectl port-foward <consul-release>-server-0 8500 &
+```
+
+-> Be sure the latest consul binary is installed locally on your host.
+[https://releases.hashicorp.com/consul/1.8.0/](https://releases.hashicorp.com/consul/1.8.0/)
+
+-> If TLS and ACLs are enabled, you must set the following environment variables.
+
+```shell-session
+$ export CONSUL_HTTP_ADDR=https://localhost:8501
+$ export CONSUL_HTTP_TOKEN=$(kubectl get secret <consul-release>-bootstrap-acl-token -o jsonpath={.data.token} | base64 -D)
+$ export CONSUL_HTTP_SSL_VERIFY=false
+```
+
+### Submitting an Ingress Gateway configuration entry
+
+Now that you have access to the Consul cli you are able to submit an ingress gateway configuration entry.
+Here is an example service definition.
+
+```shell-session
+$  cat > ingress-gateway.hcl <<EOF
+Kind = "ingress-gateway"
+Name = "ingress-gateway"
+
+Listeners = [
+ {
+   Port = 8080
+   Protocol = "http"
+   Services = [
+     {
+       Name = "static-server"
+     }
+   ]
+ }
+]
+EOF
+```
+
+Submit the ingress gateway service definition with the Consul cli using this command.
+
+```shell-session
+$ consul config write ingress-gateway.hcl
+```
+
+-> You can confirm service registration by looking at the consul-ui : http://localhost:8500/ui/dc1/services/ingress-gateway/
```

Comment:
Again, this isn't a service registration. Maybe you can say you can confirm that ingress gateways have been configured as expected by looking at.

Also if TLS is enabled then you'll need `https` and `8501`